pkg/skaffold/runner/dev.go

```diff
@@ -102,11 +102,19 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer, logger *kuber
 			instrumentation.AddDevIteration("build")
 			meterUpdated = true
 		}
-		if _, err := r.Build(ctx, out, r.changeSet.needsRebuild); err != nil {
-			logrus.Warnln("Skipping deploy due to error:", err)
+		bRes, err := r.Build(ctx, out, r.changeSet.needsRebuild)
+		if err != nil {
+			logrus.Warnln("Skipping test and deploy due to build error:", err)
 			event.DevLoopFailedInPhase(r.devIteration, sErrors.Build, err)
 			return nil
 		}
+		if !r.runCtx.SkipTests() {
```

Comment:
side note: this would be cleaner if we instantiated the `Tester` itself with the `skipTest` boolean. then we could avoid having this check here and instead have `Test` just implicitly be a noop.

would you mind adding a TODO here just mentioning this?

Reply:
Added a TODO. 
Also created a tracking issue https://github.com/GoogleContainerTools/skaffold/issues/5206.